Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ExamGroupsController.cs

```diff
@@ -0,0 +1,286 @@
+
+        public override IEnumerable GetData()
+        {
+            var examGroups = this.examGroupsData.GetAll();
+
+            if (this.User.IsLecturer())
+            {
+                examGroups = examGroups.Where(eg => eg.Contest == null ||
+                    (eg.Contest.Lecturers.Any(l => l.LecturerId == this.UserProfile.Id) ||
+                        eg.Contest.Category.Lecturers.Any(l => l.LecturerId == this.UserProfile.Id)));
+            }
+
+            return examGroups.Select(ViewModelType.FromExamGroup);
+        }
+
+        public override object GetById(object id) => this.GetByIdAsNoTracking((int)id);
+
+        [HttpPost]
+        public ActionResult Create([DataSourceRequest]DataSourceRequest request, ViewModelType model)
+        {
+            var contestId = model.ContestId.HasValue
+                ? this.contestsData.GetByIdQuery(model.ContestId.Value).Select(c => c.Id).FirstOrDefault()
+                : default(int);
+
+            if (contestId != default(int))
+            {
+                if (!this.UserHasContestRights(contestId))
+                {
+                    this.ModelState.AddModelError(nameof(model.ContestId), Resource.Cannot_attach_contest);
+                    return this.GridOperation(request, model);
+                }
+
+                this.BaseCreate(model.GetEntityModel());
+            }
+            else
+            {
+                this.ModelState.AddModelError(nameof(model.ContestId), string.Empty);
+            }
+            
+            return this.GridOperation(request, model);
+        }
+
+        [HttpPost]
+        public ActionResult Update([DataSourceRequest]DataSourceRequest request, ViewModelType model)
+        {
+            if (!model.Id.HasValue)
+            {
+                return this.GridOperation(request, model);
+            }
+
+            var entity = this.GetByIdAsNoTracking(model.Id.Value);
+
+            var examGroup = model.GetEntityModel(entity);
+
+            if (examGroup.ContestId.HasValue)
+            {
+                if (!this.contestsData
+                    .GetAll()
+                    .Any(c => c.Id == examGroup.ContestId.Value))
+                {
+                    this.ModelState.AddModelError(nameof(model.ContestId), string.Empty);
+                    return this.GridOperation(request, model);
+                }
+
+                if (!this.UserHasContestRights(examGroup.ContestId.Value))
+                {
+                    this.ModelState.AddModelError(nameof(model.ContestId), Resource.Cannot_attach_contest);
+                    return this.GridOperation(request, model);
+                }
+            }
+
+            this.BaseUpdate(examGroup);
+
+            return this.GridOperation(request, model);
+        }
+
+        [HttpPost]
+        public ActionResult Destroy([DataSourceRequest]DataSourceRequest request, ViewModelType model)
+        {
+            if (model.ContestId != null)
```

Comment:
Can this check be inverted? Something like
`If model.ContestId == null` -> error msg

Reply:
This is not for error message, there is a valid case where you want to delete an Exam Group that does not have a ContestId